pkg/api/v1alpha1/cluster_types.go

```diff
@@ -231,8 +231,8 @@ type ClusterNetwork struct {
 	Pods     Pods     `json:"pods,omitempty"`
 	Services Services `json:"services,omitempty"`
 	// CNI specifies the CNI plugin to be installed in the cluster
-	CNI CNI `json:"cni,omitempty"`
-	DNS DNS `json:"dns,omitempty"`
+	CNI CNI  `json:"cni,omitempty"`
+	DNS *DNS `json:"dns,omitempty"`
```

Comment:
What do you think about making `DNS` a struct but `ResolvConf` a pointer?
I think that express better the design of the api:
* DNS is just a grouping of configuration, it doesn't need to be a pointer. This follows the same pattern as other fields like `Networking`
* `ResolvConf` is an optional configuration for an specific sense. It makes sense that it's a pointer since that's the way the user can express they don't want to configure it.

If you decide to keep `DNS` as a pointer, then I would make `ResolvConf` a pointer as well